src/rgw/rgw_sal.h

```diff
@@ -270,7 +271,7 @@ class User {
     const RGWUserCaps& get_caps() const { return info.caps; }
     static bool empty(User* u) { return (!u || u->info.user_id.id.empty()); }
     static bool empty(std::unique_ptr<User>& u) { return (!u || u->info.user_id.id.empty()); }
-    virtual int read_attrs(const DoutPrefixProvider* dpp, optional_yield y, Attrs* uattrs, RGWObjVersionTracker* tracker = nullptr) = 0;
+    virtual int read_attrs(const DoutPrefixProvider* dpp, optional_yield y, RGWObjVersionTracker* tracker = nullptr) = 0;
```

Comment:
Is 'tracker' argument still needed? 
Like user->get_attrs(), can it be read using user->get_version_tracker() by the callers?

Reply:
tracker is an in/out parameter, so the one you pass in is read.  It may be that we can remove it and always pass the one stored in User, but I'm not sure of that.  I'll note it for future work.